### PR TITLE
fix(ui): fix navigation glitch

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-1291

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Don’t Sacrifice Readability, you are going to debug it in 6 months

File: libs/shared/ui/src/lib/components/navigation-left/navigation-left-sub-link/navigation-left-sub-link.spec.tsx [hunk 1]
```diff
@@ -10,90 +10,46 @@ jest.mock('react-router-dom', () => ({
...
-  it('should have rotate the arrow', () => {
-    render(<NavigationLeftSubLink {...props} />)
+  it('should have rotate the arrow', async () => {
+    const { userEvent } = renderWithProviders(<NavigationLeftSubLink {...props} />)
 
     const trigger = screen.getByTestId('link')
 
-    fireEvent.click(trigger)
-
+    console.log(trigger.querySelector('.icon-solid-angle-down')?.classList.toString())
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/navigation-left/navigation-left-sub-link/navigation-left-sub-link.spec.tsx:48]: need to be remove

File: libs/shared/ui/src/lib/components/error-boundary/error-boundary.tsx [hunk 2]
```diff
@@ -25,15 +24,11 @@ export function ErrorFallback({ className, error, resetErrorBoundary }: Fallback
...
+interface ErrorBoundaryProps extends PropsWithChildren, Attributes {}
 
+export function ErrorBoundary({ key, children }: ErrorBoundaryProps) {
   return (
-    <ErrorBoundaryRaw
-      key={location.pathname}
-      FallbackComponent={ErrorFallback}
-      onReset={() => window.location.reload()}
-    >
+    <ErrorBoundaryRaw key={key} FallbackComponent={ErrorFallback} onReset={() => window.location.reload()}>
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/error-boundary/error-boundary.tsx:31]: key is really necessary ? I have a warning in the console, maybe related
 
Screenshot 2024-09-11 at 17 15 27

File: libs/shared/ui/src/lib/components/navigation-left/navigation-left-sub-link/navigation-left-sub-link.tsx [hunk 3]
```diff
@@ -1,48 +1,42 @@
...
 
-export function NavigationLeftSubLink(props: NavigationLeftSubLinkProps) {
-  const { link, linkClassName, linkContent } = props
+export function NavigationLeftSubLink({ link, children }: NavigationLeftSubLinkProps) {
   const { pathname } = useLocation()
+  const isActivePath = link.subLinks.some((currentLink) =>
+    linkClassName(pathname, currentLink.url)?.includes('is-active')
+  )
+  // By default only the active path is open
+  const [open, setOpen] = useState(isActivePath)
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/navigation-left/navigation-left-sub-link/navigation-left-sub-link.tsx:23]: It's `true` by default we have all sub-links open, should be `false` 

Screenshot 2024-09-11 at 17 18 14

**Review by RemiBonnet — CHANGES_REQUESTED** [review]
Thanks @ctjhoa !

I added some comments and could you fix the storybook? `NavigationLeft` is broken

File: libs/shared/ui/src/lib/components/error-boundary/error-boundary.tsx (hunk 2 above)
Comment (ctjhoa) [reply libs/shared/ui/src/lib/components/error-boundary/error-boundary.tsx]: Fixed https://github.com/Qovery/console/pull/1660/commits/5f279de42063801fccbd75a76200084581fc3c89

File: libs/shared/ui/src/lib/components/navigation-left/navigation-left-sub-link/navigation-left-sub-link.spec.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/shared/ui/src/lib/components/navigation-left/navigation-left-sub-link/navigation-left-sub-link.spec.tsx]: Fixed https://github.com/Qovery/console/pull/1660/commits/6363b67a78181dfc46f2ffe6984a8d76594a0863

File: libs/shared/ui/src/lib/components/navigation-left/navigation-left-sub-link/navigation-left-sub-link.tsx (hunk 3 above)
Comment (ctjhoa) [reply libs/shared/ui/src/lib/components/navigation-left/navigation-left-sub-link/navigation-left-sub-link.tsx:23]: Fixed, I missed only one condition
https://github.com/Qovery/console/pull/1660/commits/97cd36bc0c8fe40ac1b911772e7e4760e0ec69fc
https://github.com/Qovery/console/pull/1660/commits/2bfca482906b9bf7e13199385a4ee9202e37a31d

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**